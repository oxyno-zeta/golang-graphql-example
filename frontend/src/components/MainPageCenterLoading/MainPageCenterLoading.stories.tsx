import React from 'react';
import { StoryFn, Meta } from '@storybook/react';
import MainPageCenterLoading, { Props } from './MainPageCenterLoading';

export default {
  title: 'Components/MainPageCenterLoading',
  component: MainPageCenterLoading,
} as Meta<typeof MainPageCenterLoading>;

const Template: StoryFn<typeof MainPageCenterLoading> = function C(args: Props) {
  return <MainPageCenterLoading {...args} />;
};

export const Playground = {
  render: Template,
  args: {},
};
