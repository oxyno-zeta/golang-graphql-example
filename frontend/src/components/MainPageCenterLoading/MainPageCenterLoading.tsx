import React from 'react';
import { Theme } from '@mui/material/styles';
import Backdrop, { BackdropProps } from '@mui/material/Backdrop';
import CircularProgress, { CircularProgressProps } from '@mui/material/CircularProgress';

export interface Props {
  backdropProps?: BackdropProps;
  circularProgressProps?: CircularProgressProps;
}

const defaultProps = {
  backdropProps: {},
  circularProgressProps: {},
};

function MainPageCenterLoading({ backdropProps, circularProgressProps }: Props) {
  return (
    <Backdrop sx={{ color: '#fff', zIndex: (theme: Theme) => theme.zIndex.drawer + 1 }} open {...backdropProps}>
      <CircularProgress color="inherit" {...circularProgressProps} />
    </Backdrop>
  );
}

MainPageCenterLoading.defaultProps = defaultProps;

export default MainPageCenterLoading;
