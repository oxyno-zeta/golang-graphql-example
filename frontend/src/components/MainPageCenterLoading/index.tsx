import MainPageCenterLoading, { Props as MainPageCenterLoadingProps } from './MainPageCenterLoading';

export default MainPageCenterLoading;
export type { MainPageCenterLoadingProps };
