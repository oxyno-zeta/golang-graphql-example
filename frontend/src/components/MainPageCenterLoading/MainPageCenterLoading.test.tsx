import React from 'react';
import { render } from '@testing-library/react';
// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';

import MainPageCenterLoading from './MainPageCenterLoading';

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key: string) => key }),
}));

describe('MainPageCenterLoading', () => {
  it('should display main page center loading', async () => {
    const { container, findByRole } = render(<MainPageCenterLoading />);

    // Find progressbar
    expect(await findByRole('progressbar', { hidden: true })).not.toBeNull();
    expect(container).toMatchSnapshot();
  });
});
