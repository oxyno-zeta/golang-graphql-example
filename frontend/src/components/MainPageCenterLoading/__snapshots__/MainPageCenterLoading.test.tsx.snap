// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPageCenterLoading should display main page center loading 1`] = `
@keyframes animation-0 {
  0% {
    -webkit-transform: rotate(0deg);
    -moz-transform: rotate(0deg);
    -ms-transform: rotate(0deg);
    transform: rotate(0deg);
  }

  100% {
    -webkit-transform: rotate(360deg);
    -moz-transform: rotate(360deg);
    -ms-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

@keyframes animation-1 {
  0% {
    stroke-dasharray: 1px,200px;
    stroke-dashoffset: 0;
  }

  50% {
    stroke-dasharray: 100px,200px;
    stroke-dashoffset: -15px;
  }

  100% {
    stroke-dasharray: 100px,200px;
    stroke-dashoffset: -125px;
  }
}

.emotion-0 {
  position: fixed;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
  background-color: rgba(0, 0, 0, 0.5);
  -webkit-tap-highlight-color: transparent;
  color: #fff;
  z-index: 1201;
}

.emotion-1 {
  display: inline-block;
  -webkit-animation: animation-0 1.4s linear infinite;
  animation: animation-0 1.4s linear infinite;
}

.emotion-2 {
  display: block;
}

.emotion-3 {
  stroke: currentColor;
  stroke-dasharray: 80px,200px;
  stroke-dashoffset: 0;
  -webkit-animation: animation-1 1.4s ease-in-out infinite;
  animation: animation-1 1.4s ease-in-out infinite;
}

<div>
  <div
    aria-hidden="true"
    class="MuiBackdrop-root  emotion-0"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  >
    <span
      class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorInherit  emotion-1"
      role="progressbar"
      style="width: 40px; height: 40px;"
    >
      <svg
        class="MuiCircularProgress-svg  emotion-2"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate  emotion-3"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </span>
  </div>
</div>
`;
